### PR TITLE
feat: ADOT Container logs collection

### PR DESCRIPTION
**Pull request by arunvthangaraj**
*Issue #, if available:* 134

*Description of changes:*

Enable container logs collection using ADOT

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/common/resources/otel-collector-config.yml [hunk 1]
```diff
@@ -1637,13 +1637,86 @@ spec:
...
+        start_at: beginning
+      {{ stop enableAdotContainerLogsCollection }}
     exporters:
       prometheusremotewrite:
         endpoint: "{{remoteWriteEndpoint}}"
         auth:
           authenticator: sigv4auth
       logging:
         loglevel: info
+      {{ start enableAdotContainerLogsCollection }}
```
Comment (elamaran11) [inline lib/common/resources/otel-collector-config.yml:1711]: On looking quickly, enableAdotContainerLogsCollection i used in two places. The logic renders only one occurence so you should use a different variable for two occurences, check out the adot collector monitoring as an example

File: lib/single-new-eks-opensource-observability-pattern/graviton-index.ts [hunk 2]
```diff
@@ -59,6 +59,12 @@ export default class SingleNewEksGravitonOpenSourceObservabilityPattern {
             "{{ stop enableAdotMetricsCollectionTelemetry }}",
             jsonStringnew.context["adotcollectormetrics.pattern.enabled"]
         );
+        doc = utils.changeTextBetweenTokens(
```
Comment (elamaran11) [inline lib/single-new-eks-opensource-observability-pattern/graviton-index.ts:62]: Please pull code from main. We had some significant changes.

File: lib/single-new-eks-opensource-observability-pattern/index.ts [hunk 3]
```diff
@@ -60,6 +60,12 @@ export default class SingleNewEksOpenSourceobservabilityPattern {
             "{{ stop enableAdotMetricsCollectionTelemetry }}",
             jsonStringnew.context["adotcollectormetrics.pattern.enabled"]
         );
+        doc = utils.changeTextBetweenTokens(
+            doc,
```
Comment (elamaran11) [inline lib/single-new-eks-opensource-observability-pattern/index.ts:64]: Please pull code from main. We had some significant changes.

**Review by elamaran11 — COMMENTED** [review]
Overall looks great. Please check on the comments in the PR. Also take care of doc and mkdocs update.

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+```
+
+!! warning This scenario might need larger worker node for the pod. 
+
+
+Once completed the rest of the _Deploying_ steps, you can move on with the deployment of the Nginx workload.
+
+## Viewing Logs in CloudWatch Logs Insights
+
+Navigate to CloudWatch, then go to "Logs Insights"
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md:60]: The pattern name is not mixed. The documentation is looking like copy paste from logs doc.

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+In the dropdown, select any of the logs that begin with "/aws/eks/single-new-eks-mixed-observability-accelerator" and run a query.
+
+Example with "kubesystem" log group:
+
+![logs-query](../images/mixed-query.png)
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md:66]: shows mixed, lets create new screenshots.

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+Then you can view the results of your query:
+
+![logs-results](../images/mixed-log-results.png)
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md:70]: shows mixed, lets create new screenshots.

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    "adotcontainerlogs.pattern.enabled": true
+  }
+```
+
+!! warning This scenario might need larger worker node for the pod. 
+
+
+Once completed the rest of the _Deploying_ steps, you can move on with the deployment of the Nginx workload.
+
+## Viewing Logs in CloudWatch Logs Insights
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md:58]: Also show the screenshot for logs before log insights.

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+![logs-query](../images/mixed-query.png)
+
+Then you can view the results of your query:
+
+![logs-results](../images/mixed-log-results.png)
+
+You can also find control plane logs in CloudWatch with Control Plane logs visible under the `/cluster` log group:
+
+![control-plane-logs](../images/control-plane-logs.png)
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md:74]: See if youc an create new screenshots.

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
+# Single Cluster Open Source Observability - NGINX Monitoring
+
+## Objective
+
+This pattern demonstrates how to use the _New EKS Cluster Open Source Observability Accelerator_ forward container logs to cloud watch using ADOT containers log collector.
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md:5]: Could you please add some narrative and also refernce the launch blog or announcement doc which we released for reInvent for this

**Review by elamaran11 — COMMENTED** [review]
@arunvthangaraj This is very nice work, thankyou for doing this. Code changes, collector config all looks great, doc needs more work and screenshots needs to be new to OSS pattern.

**Comment by elamaran11** [comment]
@arunvthangaraj Lets close the loose ends in documentation and try to get this PR to closure start of this week!

File: lib/single-new-eks-opensource-observability-pattern/graviton-index.ts (hunk 2 above)
Comment (arunvthangaraj) [reply lib/single-new-eks-opensource-observability-pattern/graviton-index.ts:82]: merged with main

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md (hunk 4 above)
Comment (arunvthangaraj) [reply docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md]: updated the documentation

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md (hunk 6 above)
Comment (arunvthangaraj) [reply docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md]: added new screen prints

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md (hunk 7 above)
Comment (arunvthangaraj) [reply docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md]: added screen prints for this

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md (hunk 9 above)
Comment (arunvthangaraj) [reply docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md]: Updated the documentation and added a link the announcement doc

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+        "GRAFANA_KUBELET_DASH_URL" : "https://raw.githubusercontent.com/aws-observability/aws-observability-accelerator/main/artifacts/grafana-dashboards/eks/infrastructure/kubelet.json",
+        "GRAFANA_NSWRKLDS_DASH_URL" : "https://raw.githubusercontent.com/aws-observability/aws-observability-accelerator/main/artifacts/grafana-dashboards/eks/infrastructure/namespace-workloads.json",
+        "GRAFANA_NODEEXP_DASH_URL" : "https://raw.githubusercontent.com/aws-observability/aws-observability-accelerator/main/artifacts/grafana-dashboards/eks/infrastructure/nodeexporter-nodes.json",
+        "GRAFANA_NODES_DASH_URL" : "https://raw.githubusercontent.com/aws-observability/aws-observability-accelerator/main/artifacts/grafana-dashboards/eks/infrastructure/nodes.json",
+        "GRAFANA_WORKLOADS_DASH_URL" : "https://raw.githubusercontent.com/aws-observability/aws-observability-accelerator/main/artifacts/grafana-dashboards/eks/infrastructure/workloads.json",
+        "GRAFANA_NGINX_DASH_URL" : "https://raw.githubusercontent.com/aws-observability/aws-observability-accelerator/main/artifacts/grafana-dashboards/eks/nginx/nginx.json"
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md:38]: remove nginx its not mandatory for your pattern option.

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+      },
+      "kustomizations": [
+        {
+          "kustomizationPath": "./artifacts/grafana-operator-manifests/eks/infrastructure"
+        },
+        {
+          "kustomizationPath": "./artifacts/grafana-operator-manifests/eks/nginx"
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-container-logs-opensource-observability.md:45]: remove nginx its not mandatory for your pattern option.

**Review by elamaran11 — CHANGES_REQUESTED** [review]
Minor feedback, rest looks good.

**Pull request merged**